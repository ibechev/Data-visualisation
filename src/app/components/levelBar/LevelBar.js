import React from "react";
import PropTypes from "prop-types";

const LevelBar = ({ level }) => {
  const LOW = "low";
  const MIDDLE = "middle";
  const VERY = "very";

  const middleBarClass = `bar ${MIDDLE} ${
    level === MIDDLE || level === VERY ? "active" : ""
  }`;
  const veryBarClass = `bar ${VERY} ${level === VERY ? "active" : ""}`;

  return (
    <div className="level-bar-wrapper">
      <p className="title">{level}</p>
      <div className="bars">
        <span className={`bar ${LOW} active`} />
        <span className={middleBarClass} />
        <span className={veryBarClass} />
      </div>
    </div>
  );
};

LevelBar.propTypes = {
  level: PropTypes.oneOf(["low", "middle", "very"]).isRequired
};

export default LevelBar;
