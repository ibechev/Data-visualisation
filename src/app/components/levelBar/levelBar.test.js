import React from "react";
import { shallow } from "enzyme";
import LevelBar from "./LevelBar";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../tests/config/testUtils";

describe("Expect <LevelBar />", () => {
  const LOW = "low";
  const MIDDLE = "middle";
  const VERY = "very";

  const defaultProps = {
    level: LOW
  };

  const setup = (testProps = {}) => {
    const props = {
      ...defaultProps,
      ...testProps
    };
    return shallow(<LevelBar {...props} />);
  };

  it("to render correctly", () => {
    const wrapper = renderer.create(<LevelBar {...defaultProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<LevelBar level={LOW} />, { level: LOW });
    propTypeErrors(<LevelBar level={MIDDLE} />, { level: MIDDLE });
    propTypeErrors(<LevelBar level={VERY} />, { level: VERY });
  });

  it("to render the correct number od <span class='bar' /> elements", () => {
    const wrapper = setup();
    expect(wrapper.find("span.bar")).toHaveLength(3);
  });

  describe("when prop 'level' is set to 'low'", () => {
    const wrapper = setup();

    it("to add class 'active' to <span class='low' />", () => {
      expect(wrapper.find("span.low").hasClass("active")).toEqual(true);
    });

    it("to not add class 'active' to <span class='middle' /> and <span class='very' />", () => {
      expect(wrapper.find("span.middle").hasClass("active")).toEqual(false);
      expect(wrapper.find("span.very").hasClass("active")).toEqual(false);
    });
  });

  describe("when prop 'level' is set to 'middle'", () => {
    const wrapper = setup({ level: MIDDLE });

    it("to add class 'active' to <span class='low' />", () => {
      expect(wrapper.find("span.low").hasClass("active")).toEqual(true);
    });

    it("to add class 'active' to <span class='middle' />", () => {
      expect(wrapper.find("span.middle").hasClass("active")).toEqual(true);
    });

    it("to not add class 'active' to <span class='very' />", () => {
      expect(wrapper.find("span.very").hasClass("active")).toEqual(false);
    });
  });

  describe("when prop 'level' is set to 'very'", () => {
    const wrapper = setup({ level: VERY });

    it("to add class 'active' to all <span class='bar' />", () => {
      expect(wrapper.find("span.low").hasClass("active")).toEqual(true);
      expect(wrapper.find("span.middle").hasClass("active")).toEqual(true);
      expect(wrapper.find("span.very").hasClass("active")).toEqual(true);
    });
  });
});
