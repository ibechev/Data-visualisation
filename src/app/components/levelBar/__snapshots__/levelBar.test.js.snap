// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LevelBar /> to render correctly 1`] = `
<div
  className="level-bar-wrapper"
>
  <p
    className="title"
  >
    low
  </p>
  <div
    className="bars"
  >
    <span
      className="bar low active"
    />
    <span
      className="bar middle "
    />
    <span
      className="bar very "
    />
  </div>
</div>
`;
